### PR TITLE
[Feature] Add AVA dataset and transformation

### PR DESCRIPTION
**Pull request by dreamerlin**

### CODE REVIEW

**Comment by kennymckormick** [comment]
@dreamerlin There is still conflicts need to be resolved

File: mmaction/datasets/ava_dataset.py [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+                 timestamp_end=1798):
+        self.exclude_file = exclude_file
+        self.label_file = label_file
+        self.proposal_file = proposal_file
+        self.filename_tmpl = filename_tmpl
+        self.num_max_proposals = num_max_proposals
+        self.timestamp_start = timestamp_start
+        self.timestamp_end = timestamp_end
+        self.logger = get_root_logger()
+        super().__init__(
```
Comment (innerlee) [inline mmaction/datasets/ava_dataset.py:100]: why put super.init in the middle?

File: mmaction/datasets/pipelines/augmentations.py [hunk 2]
```diff
@@ -86,6 +86,288 @@ def __call__(self, results):
...
+        return results
+
+    def __repr__(self):
+        repr_str = (f'{self.__class__.__name__}('
+                    f'scales={self.scales}, mode={self.mode})')
+        return repr_str
+
+
+@PIPELINES.register_module()
+class BoxRescale:
```
Comment (innerlee) [inline mmaction/datasets/pipelines/augmentations.py:156]: name proposal: `EntityBoxRescale`

File: mmaction/datasets/pipelines/augmentations.py [hunk 3]
```diff
@@ -86,6 +86,288 @@ def __call__(self, results):
...
+                scores = None
+            proposals = (proposals * img_scale).astype(np.float32)
+            results['proposals'] = proposals * scale_factor
+            results['scores'] = scores
+
+        return results
+
+
+@PIPELINES.register_module()
+class BoxCrop:
```
Comment (innerlee) [inline mmaction/datasets/pipelines/augmentations.py:195]: name

**Comment by innerlee** [comment]
looks good

File: mmaction/datasets/ava_dataset.py (hunk 1 above)
Comment (dreamerlin) [reply mmaction/datasets/ava_dataset.py:99]: we should first assign some inner attributes for `super().__init`

Comment (innerlee) [reply mmaction/datasets/ava_dataset.py:99]: why not put them in the argument list?

Comment (dreamerlin) [reply mmaction/datasets/ava_dataset.py:100]: They are commonly not used for `BaseDataset`

Comment (innerlee) [reply mmaction/datasets/ava_dataset.py:100]: if they are not used, then why they are required to be assigned beforehand?

Comment (dreamerlin) [reply mmaction/datasets/ava_dataset.py:100]: https://github.com/open-mmlab/mmaction2/blob/master/mmaction/datasets/base.py#L61-L62, and we rewrite the `self.load_annotations()`.

Comment (innerlee) [reply mmaction/datasets/ava_dataset.py:100]: ok. add a line of comment explaining this

**Review by innerlee — APPROVED** [review]

**Pull request merged**